### DISCUSSION TITLE
Check for wildcard imports

### DISCUSSION THREAD

**Discussion by JanMeier1337** [pmd/pmd#3870] [Q&A] [answered]
Hello,

we would like to disallow wildcard imports like:

`import de.webshop.*
`

Is that possible?

**Comment by oowekyala** [comment, marked as answer]
Assuming you want to analyse Java files, you can [write an XPath rule](https://pmd.github.io/latest/pmd_userdocs_extending_writing_rules_intro.html) with the following XPath expression (this is XPath 2.0):
```ruby
//ImportDeclaration[@ImportOnDemand=true()]
```
